cob_cartesian_controller/src/cartesian_controller.cpp

```diff
@@ -212,36 +212,42 @@ bool CartesianController::posePathBroadcaster(const geometry_msgs::PoseArray& ca
 
         // Send/Refresh target Frame
         transform.setOrigin( tf::Vector3(cartesian_path.poses.at(i).position.x,
-                                                 cartesian_path.poses.at(i).position.y,
-                                                 cartesian_path.poses.at(i).position.z) );
+                                         cartesian_path.poses.at(i).position.y,
+                                         cartesian_path.poses.at(i).position.z) );
         transform.setRotation( tf::Quaternion(cartesian_path.poses.at(i).orientation.x,
-                                                      cartesian_path.poses.at(i).orientation.y,
-                                                      cartesian_path.poses.at(i).orientation.z,
-                                                      cartesian_path.poses.at(i).orientation.w) );
+                                              cartesian_path.poses.at(i).orientation.y,
+                                              cartesian_path.poses.at(i).orientation.z,
+                                              cartesian_path.poses.at(i).orientation.w) );
+
         tf_broadcaster_.sendTransform(tf::StampedTransform(transform, ros::Time::now(), cartesian_path.header.frame_id, target_frame_));
 
         // Get transformation
         tf::StampedTransform stamped_transform = utils_.getStampedTransform(target_frame_, chain_tip_link_);
 
-        // Check whether chain_tip_link is within epsilon area of target_frame
-        if(!utils_.inEpsilonArea(stamped_transform, epsilon))
+        ros::spinOnce();
+        rate.sleep();
+    }
+
+    while(true)
```

Comment:
should not be `while(true)`...
maybe we can use `while (ros::ok())` or something action-related, `as_->` (http://docs.ros.org/indigo/api/actionlib/html/classactionlib_1_1SimpleActionServer.html)